lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1914,6 +1914,32 @@ def cmd_set(*args)
       print_warning("Changing the SSL option's value may require changing RPORT!")
     end
 
+    # Correctly set the file output if user provides a directory for SessionTlvLogging
+    if name.casecmp?('SessionTlvLogging')
+      log_output = nil
+      log_output_path = nil
+      if datastore[name].start_with?('file:')
+        pathname = ::Pathname.new(datastore[name].split('file:').last)
+        if ::File.directory?(pathname)
+          datastore[name] = ::File.join(datastore[name], 'sessiontlvlogging.txt')
+        end
+
+        log_output = :file
+        log_output_path = datastore[name]
+      elsif datastore[name].casecmp?('console') || datastore[name].casecmp?('true')
+        log_output = :console
+      else
+        log_output = :none
+      end
+
+      framework.sessions.each do |_index, session|
+        session.tlv_log_file_path = log_output_path
+        session.tlv_log_output = log_output
+        # After setting new sessiontlvlogging option, assume we want to retry the logging
+        session.tlv_logging_error_occured = false
```

Comment:
Let's hide this smell behind a method instead 👀

Reply:
Or if you want to reuse `initialize_tlv_logging` as a name that doesn't seem too bad to me 🤔